src/mento.ts

```diff
@@ -0,0 +1,260 @@
+import { BigNumber, ContractTransaction, Signer } from 'ethers'
+import {
+  IBroker,
+  IBroker__factory,
+  IExchangeProvider,
+  IExchangeProvider__factory,
+} from '@mento-protocol/mento-core-ts'
+import { Provider, TransactionResponse } from '@ethersproject/providers'
+import {
+  getBrokerAddressFromRegistry,
+  getSymbolFromTokenAddress,
+} from './utils'
+
+import { strict as assert } from 'assert'
+
+interface Exchange {
+  providerAddr: string
```

Comment:
Optional tip: Define a global `Address` type to be string to make it more clear when you mean for a string type be to an address. Example: https://github.com/hyperlane-xyz/hyperlane-warp-ui-template/blob/main/src/global.d.ts

Reply:
I like that!